src/hist/plot.py

```diff
@@ -210,159 +223,334 @@ def plot2d_full(
+    if coverage is None:
+        coverage = stats.norm.cdf(1) - stats.norm.cdf(-1)
+    scale = np.empty_like(values)
+    scale[values != 0] = variances[values != 0] / values[values != 0]
+    if np.sum(values == 0) > 0:
+        missing = np.where(values == 0)
+        available = np.nonzero(values)
+        if len(available[0]) == 0:
+            raise RuntimeWarning(
+                "All values are zero! Cannot compute meaningful error bars",
+            )
+        nearest = np.sum(
+            [np.square(np.subtract.outer(d, d0)) for d, d0 in zip(available, missing)]
+        ).argmin(axis=0)
+        argnearest = tuple(dim[nearest] for dim in available)
+        scale[missing] = scale[argnearest]
+    counts = values / scale
+    interval_min = scale * stats.chi2.ppf((1 - coverage) / 2, 2 * counts) / 2.0
+    interval_max = scale * stats.chi2.ppf((1 + coverage) / 2, 2 * (counts + 1)) / 2.0
+    interval = np.array([interval_min, interval_max])
+    interval[interval == np.nan] = 0.0  # chi2.ppf produces nan for counts=0
+    return interval
+
+
+def clopper_pearson_interval(
+    num: np.ndarray, denom: np.ndarray, coverage: "Optional[float]" = None
+) -> np.ndarray:
+    """Compute Clopper-Pearson coverage interval for a binomial distribution
+    Parameters
+    ----------
+        num : numpy.ndarray
+            Numerator, or number of successes, vectorized
+        denom : numpy.ndarray
+            Denominator or number of trials, vectorized
+        coverage : float, optional
+            Central coverage interval, defaults to 68%
+    c.f. http://en.wikipedia.org/wiki/Binomial_proportion_confidence_interval
+    """
+    if coverage is None:
+        coverage = stats.norm.cdf(1) - stats.norm.cdf(-1)
+    # Numerator is subset of denominator
+    if np.any(num > denom):
+        raise ValueError(
+            "Found numerator larger than denominator while calculating binomial uncertainty"
         )
-    yerr = np.sqrt(variances)
-
-    if isinstance(func, str):
-        if func in {"gauss", "gaus"}:
-            # gaussian with reasonable initial guesses for parameters
-            constant = float(ydata.max())
-            mean = (ydata * xdata).sum() / ydata.sum()
-            sigma = (ydata * (xdata - mean) ** 2.0).sum() / ydata.sum()
-
-            def func(
-                x: np.ndarray,
-                constant: float = constant,
-                mean: float = mean,
-                sigma: float = sigma,
-            ) -> np.ndarray:
-                return constant * np.exp(-((x - mean) ** 2.0) / (2 * sigma ** 2))  # type: ignore
-
-        else:
-            func = _expr_to_lambda(func)
-
-    assert not isinstance(func, str)
-
-    parnames = list(inspect.signature(func).parameters)[1:]
+    interval_min = stats.beta.ppf((1 - coverage) / 2, num, denom - num + 1)
+    interval_max = stats.beta.ppf((1 + coverage) / 2, num + 1, denom - num)
+    interval = np.array([interval_min, interval_max])
```

Comment:
Is this actually a np.stack?

Reply:
Yup. This was [a copypaste from `coffea`](https://github.com/CoffeaTeam/coffea/blob/12228d5564963234434d5c881efa761e088c298a/coffea/hist/plot.py#L75), but a good catch and example of some things I might want to clean up in those.